topi/python/topi/x86/conv2d.py

```diff
@@ -81,81 +81,153 @@ def _declaration_conv(data, kernel, stride, padding, layout, out_dtype):
         raise ValueError("not support this layout {} yet".format(layout))
 
 
+def _traverse_conv2d(op, conv_tag, default_schedule):
+    """Traverse operators from computation graph"""
+    # inline all one-to-one-mapping operators except the last stage (output)
+    if tag.is_broadcast(op.tag):
+        if op not in s.outputs:
+            s[op].compute_inline()
+        else: # inject custom schedule
+            if len(op.axis) == 4 and 'avx' not in str(target): # schedule bias + bn + relu
+                n, c, h, w = op.axis
+                fused = s[op].fuse(n, c)
+                s[op].parallel(fused)
+                s[op].vectorize(w)
+        for tensor in op.input_tensors:
+            if tensor.op.input_tensors:
+                _traverse_conv2d(tensor.op, conv_tag, default_schedule)
+
+    if op.tag == conv_tag:
+        default_schedule(op)
+
+
+def _conv2d_default_schedule(s, conv, data, layout='nchw'):
+    """NCHW conv2d schedule for non imagenet workloads"""
+    data_pad = None
+    if isinstance(data.op, tvm.tensor.ComputeOp) and "pad" in data.op.tag:
+        data_pad = data
+        data = data_pad.op.input_tensors[0]
+    elif isinstance(data.op, tvm.tensor.PlaceholderOp):
+        return
+
+    _schedule_pad(s, data_pad, layout)
+    n, c, h, w = conv.op.axis
+    rc, ry, rx = conv.op.reduce_axis
+    fused = s[conv].fuse(n, c)
+    s[conv].parallel(fused)
+    wo, wi = s[conv].split(w, factor=16)
+    s[conv].reorder(fused, rc, h, wo, ry, rx, wi)  # move rc to outer loop
+    s[conv].unroll(rx)
+    s[conv].unroll(ry)
+    s[conv].vectorize(wi)
+
+
+def _get_axes_nchw(tensor, layout):
+    """Returns operator axes as [N, C, H, W].
+
+    Parameters
+    ----------
+    tensor: The tensor
+    layout: should be a string containing each of the characters {n, c, h, w}
+    """
+    return [tensor.op.axis[i] for oa in 'nchw'
+            for i, ia in enumerate(layout) if oa == ia]
+
+
+def _schedule_pad(schedule, padded_tensor, layout='nchw'):
+    """Fuse and parallelize a padding op."""
+    n, c, h, w = _get_axes_nchw(padded_tensor, layout)
+    pad_fused = schedule[padded_tensor].fuse(n, c)
```

Comment:
not sure, can we fuse two nonadjacent axis? if run `_get_axes_nchw(padded_tensor, 'nhwc')`, `n` and `c` will be nonadjacent, right?